Keyvisual CTA component, checkmarks and height correction

bozojovicic opened the pull request:
Fixes #

Checkmarks added to the commercial-hero-banner
The icon check-circle from src/components/00-materials/icons-raw/material-design/check_circle.svg used for checkmarks

<img width="510" alt="checkmarks" src="https://user-images.githubusercontent.com/37440641/136000714-00f532b2-44e5-4a1a-a25c-246fb91978c0.png">

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch-webhub-cloud/pattern-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [ ] Tests are sufficient.
- [ ] Modifications still work in IE.
- [ ] Documentation is up to date.
- [ ] Changelog is up to date.
- [ ] Typescript typings for properties are up to date.
- [ ] Designers approved changes or no approval needed.
- [ ] Dependencies to other components still work.

Commit:
Keyvisual CTA component, checkmarks and height correction

Ben-Zahler commented on src/components/30-organisms/commercial-hero-banner/story.js:
```diff
@@ -32,6 +32,7 @@ export const CommercialHeroBanner = () => {
 
   const wrapper = document.createElement('div');
   const template = html`
+    <div class="m-landingpage-keyvisual">
```
@bozojovicic do we really need this? I don't think that class is part of the patterns lib code

bozojovicic replied:
You are right, I removed it.

Commit:
Keyvisual CTA component, checkmarks and height correction

raphaellueckl commented on src/components/30-organisms/commercial-hero-banner/story.js:
```diff
@@ -55,6 +58,40 @@ export const CommercialHeroBanner = () => {
         >GET A QUOTE</axa-button-link
       >
     </axa-commercial-hero-banner>
+    <br />
+    <br />
+    <axa-commercial-hero-banner
+      variant="${variant}"
+      imagesource="${imageSource}"
+    >
+      <header slot="title">
+        <p>This example shows specific picture classes</p>
+        <h1>Drive with peace of mind</h1>
+      </header>
+      <div slot="content" class="checkmarks">
+        <div class="checkmark">
+          <axa-icon class="checkmark-icon" icon="check-circle" size="auto"></axa-icon>
+          <span class="checkmark-text">Extra Cookie</span>
+        </div>
+        <div class="checkmark">
+          <axa-icon class="checkmark-icon" icon="check-circle" size="auto"></axa-icon>
+          <span class="checkmark-text">Awesome People</span>
+        </div>
+        <div class="checkmark">
+          <axa-icon class="checkmark-icon" icon="check-circle" size="auto"></axa-icon>
+          <span class="checkmark-text">Best in class Service</span>
+        </div>
+      </div>
+      
+      <axa-button-link
+        href="https://axa.ch"
+        slot="button"
+        variant="red"
+        size="large"
+        >GET A QUOTE</axa-button-link
+      >
+    </axa-commercial-hero-banner>      
+    </div>
```
Having 2 elements within the same story is a bit weird. The main story should not be polluted with several variants. Better use your new one as main and adapt the tests.

bozojovicic replied:
I used the new one and removed from test the parts related to disclaimer. All tests pass now.

bozojovicic replied:
I also used prettier to format the code in commercial-hero-banner/story.js but then I realised that you already did that and pushed.

raphaellueckl reviewed (changes requested):
Looks good overall. But there are no tests so far. You can add some that validate your functionality in the `pw.test.ui.js` file within the commercial-hero-banner directory.

Ben told me I should review, not sure if I'm interrupting here.

raphaellueckl committed:
prettify

raphaellueckl commented on src/components/30-organisms/commercial-hero-banner/child.scss:
```diff
@@ -13,13 +13,27 @@ axa-commercial-hero-banner {
   }
 
+  .checkmark {
+    margin-bottom: 5px;
```
If you would make `.checkmark` with `display: flex; justify-content: center;` ....

raphaellueckl commented on src/components/30-organisms/commercial-hero-banner/child.scss:
```diff
@@ -13,13 +13,27 @@ axa-commercial-hero-banner {
+
+  .checkmark-text {
+    position: relative;
+    top: -5px;
```
...you could get rid of this negative positioning. I'm not a fan of negative sizing information, as it's coupled to font and font-size.

bozojovicic replied:
I did that and this is how it looks like now :
<img width="436" alt="Screenshot 2021-10-05 at 16 26 39" src="https://user-images.githubusercontent.com/37440641/136049293-205df474-000e-4deb-a45b-da81a275c7ff.png">
Properties like justify-content, align-items etc. makes no difference since it is already centralized but in my opinion the text is a bit too much moved to the top, for 1-2 pixels.
Take a look and let me know it it's ok.

raphaellueckl replied:
Weird, in my chromium on macos it looks perfect:

![image](https://user-images.githubusercontent.com/8248662/136756512-be54d6d5-f609-4722-bcb9-9e546b696b71.png)

Commit:
Keyvisual CTA component, checkmarks and height correction

raphaellueckl commented:
Mobile view is broken now: 
![image](https://user-images.githubusercontent.com/8248662/136756755-047a7cab-07a1-4a51-a5be-1c8fa257801f.png)

raphaellueckl commented on src/components/30-organisms/commercial-hero-banner/story.js:
```diff
@@ -32,29 +32,51 @@ export const CommercialHeroBanner = () => {
+          </div>
+          <div class="checkmark">
+            <axa-icon
+              class="checkmark-icon"
```
Those new classes have to be described in the readme, as we will style code that comes with those classes.

bozojovicic replied:
done

raphaellueckl reviewed (changes requested):
Hi Bozo

Looks good, but please don't just remove checks that we used in tests. Also see my otehr comments.

But overall, sweet work!

Commit:
Keyvisual CTA component, checkmarks and height correction

Commit:
Keyvisual CTA component, checkmarks and height correction

bozojovicic commented:
@raphaellueckl All comments addressed. I don't know why I removed those tests. I added them back.
Mobile view is not broken, the width of that white space is determined by the width of the text inside and since the long description is now replaced with short checkmarks it looks like it is broken. I have set width to 100% and now it should be OK even for short text in both text description and checkmarks.

Commit:
Keyvisual CTA component, checkmarks and height correction

Commit:
Keyvisual CTA component, checkmarks and height correction

Commit:
prettify

Commit:
prettify

raphaellueckl reviewed (approved):
nice work man!
